src/main/java/uk/gov/companieshouse/api/accounts/controller/DirectorsController.java

```diff
@@ -0,0 +1,145 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.links.SmallFullLinkType;
+import uk.gov.companieshouse.api.accounts.model.rest.DirectorsReport;
+import uk.gov.companieshouse.api.accounts.model.rest.directorsreport.Director;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.impl.DirectorService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
+import uk.gov.companieshouse.api.accounts.utility.ErrorMapper;
+import uk.gov.companieshouse.api.accounts.utility.LoggingHelper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestController
+@RequestMapping("/transactions/{transaction_id}/company-accounts/{company_accounts_id}/small-full/directors-report/directors")
+public class DirectorsController {
+
+    @Autowired
+    private DirectorService directorService;
+
+    @Autowired
+    private ApiResponseMapper apiResponseMapper;
+
+    @Autowired
+    private ErrorMapper errorMapper;
+
+
+    @PostMapping
+    public ResponseEntity create(@RequestBody Director director, BindingResult bindingResult,
+                                 @PathVariable("companyAccountId") String companyAccountId, HttpServletRequest request) {
+
+        if (bindingResult.hasErrors()) {
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<Director> response =
+                    directorService.create(director, transaction, companyAccountId, request);
+
+            return apiResponseMapper
+                    .map(response.getStatus(), response.getData(), response.getErrors());
+
+        } catch (DataException ex) {
+
+            LoggingHelper.logException(companyAccountId, transaction,
+                    "Failed to create a Director resource", ex, request);
+            return apiResponseMapper.getErrorResponse();
+        }
+    }
+
+    @GetMapping
```

Comment:
Since we're fetching a particular director, this should be `@GetMapping("/{directorId}")` - the value set in this annotation is appended to the request mapping at the top of the class, so GET requests against:
```
/transactions/{transaction_id}/company-accounts/{companyAccountId}/small-full/directors-report/directors/{directorId}
```
will map to this method.
The same is relevant to PUT and DELETE, since for these 3 operations we're manipulating a particular resource.
For the PUT, the `directorId` is then accessible as a PathVariable with which to perform the logic suggested in a previous comment.